doc/Programming-guide/Your-first-Kaa-application/index.md

```diff
@@ -2,7 +2,615 @@
+temperature data from a sensor. Usually, simple sensors stream data at constant
+rate, no matter what. However, rising demand of mobility and power efficiency dictates own
+rules of game. It is not viable to keep sensors streaming all the time, especially
+if a data is not required at this moment. That's a point where we can save power by
+configuring sample period.
+
+To satisfy application requirements two of main Kaa features are used:
+
+- **Data Collection** feature that allows to send data from endpoints
+to the Kaa server. In this example Data Collection is used to reliably transmit
+temperature samples from sensor to the Kaa.
+
+- **Configuration** feature that allows to broadcast configuration parameters
+from the Kaa server to Kaa endpoints. In this guide Configuration is used
+to pass a sample period of the temperature to endpoint.
+
+# Adding application
+
+To add an application, proceed as follows:
+
+1. Open the [Kaa admin UI in your browser](http://127.0.0.1:8080) and log in
+as a tenant administrator (user/password: admin/admin123).
+1. Select **Applications** on the navigation panel on the left side and, in the
+Applications window that opens, click **Add application**.
+1. In the Add application window, enter the application name and then click **Add**.
+For this guide we will use **Trustful** credential service and our application will be
+named "My First Kaa Application"
+
+    ![Admin Console](attach/new_app.png)
+
+After the application has been added, you may log out. We will not be using
+the tenant administrator role in this guide anymore.
+
+# Creating schemas
+
+The application that you have created in the previous step already includes
+the default versions of the profile, configuration, notification and log schemes
+ready for use. However, in this sample application, we will use a custom data
+collection and configuration schemas for demonstration purposes. Default value for
+sampling period is set to 1 second.
+
+To create and upload the schemas, proceed as follows:
+
+1. Create the *data-schema.json* file on your PC with the following schema definition:
+
+        {
+            "type": "record",
+            "name": "DataCollection",
+            "namespace": "org.kaaproject.kaa.schema.example",
+            "fields": [
+                {
+                    "name": "temperature",
+                    "type": "int"
+                }
+            ]
+        }
+1. Create the *configuration-schema.json* file and add following schema:
+
+        {
+            "type": "record",
+            "name": "Configuration",
+            "namespace": "org.kaaproject.kaa.schema.example",
+            "fields": [
+                {
+                    "name": "samplePeriod",
+                    "type": "int",
+                    "by_default": "1"
+                }
+            ]
+        }
+1. Open the admin UI in your browser and log in as
+    a tenant developer (user/password: devuser/devuser123).
```

Comment:
Again, the use of the credentials should be explained.
By my first-look opinion, they are hard-coded into the system. Is that true?